core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -301,6 +303,9 @@ private Schema lazyColumnProjection() {
       requiredFieldIds.addAll(selectedIds);
 
       return TypeUtil.select(table.schema(), requiredFieldIds);
+    } else if (context.projectedSchema() != null) {
+
+      return context.projectedSchema();
```

Comment:
Nit: I find this more readable if the blank line is before the `else` to distinguish between the previous block and the else case.